src/NServiceBus.RavenDB/Outbox/OutboxPersister.cs

```diff
@@ -53,7 +56,17 @@ public OutboxPersister(IDocumentStore documentStore, string endpointName)
 
         public Task<OutboxTransaction> BeginTransaction(ContextBag context)
         {
-            var session = documentStore.OpenAsyncSession();
+            IAsyncDocumentSession session;
+
+            var receiveContext = context as ITransportReceiveContext;
```

Comment:
We should check with Core if this can be something other than `ITransportReceiveContext`, and probably I'd prefer to throw if it's not a `ITransportReceiveContext`

Reply:
So, from unit tests, it's just a raw ContextBag. That's why I changed it to look like this rather than a direct cast.

Reply:
Any way we can mock `ITransportReceiveContext` in tests?